### PR TITLE
project idiyanale: multi-token community contributions for verified creators milestone 1

### PR DESCRIPTION
**Pull request by woss**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/1224

### CODE REVIEW

File: deliveries/anagolay-project-idiyanale-multi-token-community-contributions-for-verified-creators-milestone-1.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Milestone Delivery :mailbox:
+
+**The [invoice form :pencil:](https://docs.google.com/forms/d/e/1FAIpQLSfmNYaoCgrxyhzgoKQ0ynQvnNRoTmgApz9NrMp-hd8mhIiO0A/viewform) has been filled out correctly for this milestone and the delivery is according to the official [milestone delivery guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).**
+
+- **Application Document:** [Application file from the merged PR](https://github.com/anagolay/Grants-Program/blob/1eb5a49ca0e98f81f684c6af2d2d455b6bbcf0da/applications/anagolay-project-idiyanale-multi-token-community-contributions-for-verified-creators.md)
```
Comment (alxs) [inline deliveries/anagolay-project-idiyanale-multi-token-community-contributions-for-verified-creators-milestone-1.md:5]:
```suggestion
- **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/anagolay-project-idiyanale-multi-token-community-contributions-for-verified-creators.md
```

**Review by alxs — CHANGES_REQUESTED** [review]

**Comment by alxs** [comment]
Thanks for the delivery @woss. We'll look into it as soon as possible.

**Comment by dsm-w3f** [comment]
@woss thank you for the milestone submission. Please see [the evaluation document](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/anagolay-project-idiyanale-multi-token_1_dsm-w3f.md) and provide proper answers and fixes. After that, let me know to continue with the evaluation.

**Comment by woss** [comment]
Hi @dsm-w3f ,

> If I could run it locally this balance could be supplied by test accounts, such as Alice, Bob, etc.
Nothing is stopping you to run the chain based on the image we provided. If you prefer not to do it let me know the address and i will transfer some IDI so you can use testnet. 

> 0d broken link
This is not the case. I have made a comment about this. and the screenshot is here so you have to use it in the compose file or create a container with this. 
image

For the rest of the comments, I somewhat agree and on some, i do not. For example, the documentation warning are mostly irrelevant since they are not supposed to represent the actual links, rather than `what can it look like`.  

On the matter of code coverage, the documentation does not specify what is the minimum threshold so the tests are considered to cover the code. We wrote the tests to cover the most important parts and since this is the first version of the pallet and the pallet code is subject to change some parts will not be there in a few months and some will change so will tests and features. I suggest if you ( as an evaluator ) will be focusing on the code coverage the minimum level should be included in the guides of the grant.

**Comment by dsm-w3f** [comment]
@woss thank you for the answers. Regarding docker, we use it for testing the software in a easier/isolated way. However, proper instructions for running it should be provided. I tried to pull an image from the link that you posted and didn't worked for me. Can you please include in the tutorial for using the software the instructions to use this image?

The ideal scenario for us is that we can run all parts of the software locally and build by the source code, using docker or not. 

Regarding code quality, we always evaluate the quality of the delivery code. We have many different kind of projects, and no specific code coverage is set in the guidelines. However, stakeholders and users usually expect high quality software and all information from the evaluation is public. Feel free to chose if you want to improve code quality. This is not mandatory, is more in the sense of helping grantees to improve their software.

**Comment by woss** [comment]
@dsm-w3f ,

not sure why you cannot pull the docker image, I just tested it, and it works as expected, so the potential issue is on your end.
image

The doc is updated with the instructions that require copy/paste ( docker-compose.yml ), which i hope you will find easy to use. 

Also, I have added a section that will help you to build the app and test it in the browser. The commands in the snippet are working, so you shouldn't have any issues; if you do, then I suggest opening the anagolay js SDK in gitpod and testing there.

**Comment by dsm-w3f** [comment]
@woss thank you for the improvements. I was able to run the system locally, however, when I tried to verify a domain I got this error:

image

I seems to be related to balance. I have 8 MIDI in Alice Test account. Is this enough? Can you assist to bypass this error? Please also notice the undefined value in the value (content) field, this appeared before the transaction confirmation windows open.

**Comment by woss** [comment]
@dsm-w3f if you look at the bottom left corner, you will see that you didn't connect to the local chain regardless of you running the application locally. it defaults to the testnet. to solve this, you have 2 choices. replace the default `wss` location, then recompile or use the account switcher in the navbar to connect to the local chain ( this works the same as polkadotjs app ). This will solve the issues with the Alice account and the balances.

`undefined`, that indeed is a potential bug, but that is only the UI state, which probably svelte didn't handle correctly. it is insignificant for your case, I will make an internal bug report so we can track it. 

Let me know if you need anything else.

**Comment by dsm-w3f** [comment]
@woss thank you for the help. I updated the evaluation document. Now worked as in the video tutorial. Let me know if you intend to perform the code quality improvements or not for I finish the evaluation.

**Comment by woss** [comment]
> @woss thank you for the help. I updated the evaluation document. Now worked as in the video tutorial. Let me know if you intend to perform the code quality improvements or not for I finish the evaluation.

that's great news.

regarding the code quality improvements, I really don't see the real point in doing it at this stage since this is the first version of the pallet and a subject to change. if it is really important to you to have that we will do it, otherwise we will not. 

Thank you for your effort and time, i already reported the bug and made internal issues based on your comments so we get reminded about the issues when we start second iteration on the `verification` pallet.

**Comment by dsm-w3f** [comment]
@woss regarding code quality improvements they are important, would be nice to perform them for the next milestone delivery. Ultimately , they can affect the user experience with the software. The milestone was approved. I'll forward internally your invoice and it should be payed within two weeks (could happen before). Good Job!

**Comment by woss** [comment]
They will be included in the next milestone, if we don't change anything in the pallet. Thank you for your time and the evaluation.

**Comment by RouvenP** [comment]
hi @woss we're currently switching from USDT on Ethereum to Polkadot/Kusama. Would you be open to receive USDT on Polkadot(Statemint) or Kusama(Statemine)?

**Comment by woss** [comment]
Hi @RouvenP , what would that imply? Do we need to change the address and if yes, can you point me where to do it. We are currently using the finance for this grant address

**Comment by woss** [comment]
@RouvenP after the discussion with the SBP guys, we made a decision to stick with the current setup for this grant.

**Comment by RouvenP** [comment]
hi @woss thanks for the feedback. We will proceed with ETH-USDT.

**Comment by Eltairovy** [comment]
Hi @RouvenP would you mind specifying the timing, when can we expect the payment?
Many thanks.

**Comment by Eltairovy** [comment]
Hi @RouvenP @alxs, please let us know regarding the payment time. It's been 2 weeks since the milestone completion,  in case of a further delay we'll need to re-plan accordingly. Thank you.

**Pull request merged**